.devcontainer/Dockerfile

```diff
@@ -17,12 +17,15 @@ ARG USER_GID=$USER_UID
 
 # setup apt / certificates
 RUN apt-get update \
-    && apt-get -y install --no-install-recommends apt-utils dialog ca-certificates 2>&1
+    && apt-get -y install --no-install-recommends ca-certificates 2>&1
 
 # use mirrors instead of hardcoded location for performance
-RUN sed -i -e 's/http:\/\/us.archive/mirror:\/\/mirrors/' -e 's/\/ubuntu\//\/mirrors.txt/' /etc/apt/sources.list \
+ARG USE_APT_MIRRORS=0
+RUN if [ "${USE_APT_MIRRORS}" = "1" ]; then \
+    sed -i -e 's/http:\/\/us.archive/mirror:\/\/mirrors/' -e 's/\/ubuntu\//\/mirrors.txt/' /etc/apt/sources.list \
     && sed -i -e 's/http:\/\/security/mirror:\/\/mirrors/' -e 's/\/ubuntu\//\/mirrors.txt/' /etc/apt/sources.list \
-    && sed -i -e 's/http:\/\/archive/mirror:\/\/mirrors/' -e 's/\/ubuntu\//\/mirrors.txt/' /etc/apt/sources.list
+    && sed -i -e 's/http:\/\/archive/mirror:\/\/mirrors/' -e 's/\/ubuntu\//\/mirrors.txt/' /etc/apt/sources.list \
+    ; fi
```

Comment:
Another idea, that might be better, would be for the build-arg to provide the mirror URL that should be used, that way people can choose their mirror.

Reply:
this seems fine as it's the recommended way that automatically picks the right one based on location

Reply:
or are you talking about the case when  `USE_APT_MIRRORS` is false? That may make sense: right now configs use `archive.ubuntu.com` but I noticed (on my machine) that if I force it to `us.archive.ubuntu.com` instead it's about twice as fast (but still slower than other mirrors).

This may make a big difference for people not in the US.

Reply:
I was meaning change the var to something like `APT_MIRROR` which you can configure to `http://mirrors.ubuntu.com/mirrors.txt`, or you could manually specify your country with `http://mirrors.ubuntu.com/AU.txt`, or you can select a specific mirror if for example you're on a University campus, or in AWS, where both typically have local network mirrors. This would be helpful on AWS specifically.